File: backend/Models/Billing.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public class Billing
    {
        public int Id { get; set; }

        [Required]
        public int PatientId { get; set; }

        public Patient? Patient { get; set; }

        [Required]
        public DateTime BillingDate { get; set; }

        [Required]
        public string ServiceName { get; set; } = string.Empty;

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        public string PaymentStatus { get; set; } = "Pending";

        public string PaymentMethod { get; set; } = string.Empty;

        public string Notes { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: frontend/src/app/models/billing.ts
import { Patient } from './patient';

export interface Billing {
  id: number;
  patientId: number;
  patient?: Patient;
  billingDate: string;
  serviceName: string;
  amount: number;
  paymentStatus: string;
  paymentMethod: string;
  notes: string;
  createdAt: string;
}

export interface CreateBilling {
  patientId: number;
  billingDate: string;
  serviceName: string;
  amount: number;
  paymentStatus: string;
  paymentMethod: string;
  notes: string;
}